packages/components/src/custom-select-control-v2/legacy-component/index.tsx

```diff
@@ -36,24 +36,23 @@ function CustomSelectControl( props: LegacyCustomSelectProps ) {
 			await Promise.resolve();
 			const state = store.getState();
 
+			const option = options.find( ( item ) => item.name === nextValue );
+
 			const changeObject = {
 				highlightedIndex: state.renderedItems.findIndex(
 					( item ) => item.value === nextValue
 				),
 				inputValue: '',
 				isOpen: state.open,
-				selectedItem: {
-					name: nextValue as string,
-					key: nextValue as string,
-				},
+				selectedItem: option!,
```

Comment:
Instead of forcing `option` to be non-nullable (which may cause unintended bugs in the future), we may want to add a type guard / early return. I'll see if that is a possibility here